sui/tests/shared_objects_tests.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use bytes::Bytes;
+use futures::{sink::SinkExt, stream::StreamExt};
+use sui::config::AuthorityPrivateInfo;
+use sui_types::error::SuiError;
+use sui_types::messages::ConsensusTransaction;
+use sui_types::serialize::SerializedMessage;
+use sui_types::serialize::{deserialize_message, serialize_consensus_transaction};
+use test_utils::authority::{spawn_test_authorities, test_authority_configs};
+use test_utils::messages::test_shared_object_certificates;
+use test_utils::objects::{test_gas_objects, test_shared_object};
+use tokio::net::TcpStream;
+use tokio_util::codec::Framed;
+use tokio_util::codec::LengthDelimitedCodec;
+
+/// Submits a transaction to a Sui authority.
+async fn submit_transaction(
+    transaction: Bytes,
+    config: &AuthorityPrivateInfo,
+) -> SerializedMessage {
+    let authority_address = format!("{}:{}", config.host, config.port);
+    let stream = TcpStream::connect(authority_address).await.unwrap();
+    let mut connection = Framed::new(stream, LengthDelimitedCodec::new());
+
+    connection.send(transaction).await.unwrap();
+    let bytes = connection.next().await.unwrap().unwrap();
+    deserialize_message(&bytes[..]).unwrap()
+}
+
+#[tokio::test]
+async fn shared_object_transaction() {
```

Comment:
Are we testing any "silly" attacks like trying to double-spend?

Reply:
Not yet, this is for a next PR. I still don't know how to make a shared-object transaction that calls an actual smart contract

Reply:
This test simply tests the flow: Client -> Sui -> consensus -> Sui -> client

Reply:
> Not yet, this is for a next PR. I still don't know how to make a shared-object transaction that calls an actual smart contract

Make an issue, assign it to Xun + Move team, we need this ASAP for devnet.

Reply:
Ha do we need this for DevNet? I was pushing very hard because I wanted all the infra in the `main` branch before DevNet so that I can focus on checkpoint/reconfiguration. To support shared objects in DevNet we need to adapt the Gateway and the wallet I believe.